simd/src/Kokkos_SIMD.hpp

```diff
@@ -85,32 +85,39 @@ namespace Impl {
 #if defined(KOKKOS_ARCH_AVX512XEON)
 template <class T>
 using host_fixed_native = avx512_fixed_size<8>;
-template <int N>
-using host_native_abi = avx512_fixed_size<N>;
+template <typename T, int N>
+using host_native_abi =
+    std::conditional_t<N == 0, avx512_fixed_size<512 / sizeof(T)>,
+                       avx512_fixed_size<N>>;
 
 #elif defined(KOKKOS_ARCH_AVX2)
 template <class T>
 using host_fixed_native = avx2_fixed_size<4>;
-template <int N>
-using host_native_abi = avx2_fixed_size<N>;
+template <typename T, int N>
+using host_native_abi =
+    std::conditional_t<N == 0, avx2_fixed_size<256 / sizeof(T)>,
+                       avx2_fixed_size<N>>;
 
 #elif defined(KOKKOS_ARCH_ARM_SVE)
 template <class T>
 using host_fixed_native =
     sve_fixed_size<(__ARM_FEATURE_SVE_BITS / (8 * sizeof(T)))>;
-template <int N>
-using host_native_abi = sve_fixed_size<N>;
+template <typename T, int N>
+using host_native_abi =
+    std::conditional_t<N == 0, host_fixed_native<T>, sve_fixed_size<N>>;
 
 #elif defined(KOKKOS_ARCH_ARM_NEON)
 template <class T>
 using host_fixed_native = neon_fixed_size<2>;
-template <int N>
-using host_native_abi = neon_fixed_size<N>;
+template <typename T, int N>
+using host_native_abi =
+    std::conditional_t<N == 0, avx512_fixed_size<128 / sizeof(T)>,
+                       neon_fixed_size<N>>;
```

Comment:
There is a typo here, should be `neon` instead of `avx512`
```suggestion
using host_native_abi =
    std::conditional_t<N == 0, neon_fixed_size<128 / sizeof(T)>,
                       neon_fixed_size<N>>;
```

Reply:
Good catch. Fixed.